Dodge.js

```diff
@@ -0,0 +1,179 @@
+import { StatusBar } from 'expo-status-bar';
+import { StyleSheet, Text, View, Dimensions, Button, Image } from 'react-native';
+import { Knight, Circle, Brick, Grenade, MoveKnight, Shrapnel, counter} from "./Entities.js";
+import { GameEngine } from "react-native-game-engine";
+import Matter from "matter-js";
+import AxisPad from 'react-native-axis-pad';
+////this looks interesting for the attacker placing/throwing projectiles:
+//https://reactnative.dev/docs/panresponder 
+//https://medium.com/yapsody-engineering/hooks-and-function-components-in-react-native-d4e667c90cda
+var ents = {}
+
+const generateUniqueId = require('generate-unique-id');
+
+const MAXwidth = Dimensions.get('window').width;
+const MAXheight = Dimensions.get('window').height;
+
+
+//adds an entity given its type, if it's a Grenade set a timer for explosion
+function AddEntity(entity){
+let id = generateUniqueId()
+ents[id] = new entity(id)
```

Comment:
This is interesting, but it would probably be better for this function to accept an already constructed entity. i.e.

In AddEntity function:
```
function AddEntity(entity) {
  // Assume all entities have an id
  ents[entity.id] = entity; // Just store the value
}
```

In the entity class definition:
```
class Grenade {
  constructor() {
     // In my opinion: Nothing wrong for entities not in the list to have ids, maybe even useful for debugging.
     this.id = generateUniqueId();
  }
}
```

Elsewhere, when you add an entity:
```
// Make a new grenade, and keep track of it.
AddEntity(new Grenade(50, 100));
```

Reply:
This gives you some flexibility too, e.g. different classes could have different styles of unique IDs for readability:
```
class Knight {
  constructor() {
    // As long as there really is only one "Knight", we can make it easy to lookup
    // Elsewhere: ents["knight"] will refer to this object.
    this.id = "knight"
  }
}
```

```
class Grenade {
  constructor() {
     // Still unique, but it's easier to tell what kind of object it is at a glance.
     // It's probably not good to rely on this for anything but debugging.
     // If you want to tell what type of object something is, I suggest having another field for that explicitly.
     this.id = "grenade-" + generateUniqueId();
  }
}
```